### PR TITLE
[beta] Prepare beta 1.42.0

### PR DESCRIPTION
**Pull request by emilyalbini**
r? @ghost

### CODE REVIEW

**Comment by Centril** [comment]
@pietroalbini Can you add https://github.com/rust-lang/rust/pull/68586 to the PR so that we can avoid having to beta-backport it?

**Comment by mati865** [comment]
I think https://github.com/rust-lang/rust/pull/68579 should also be included.
Otherwise Cargo from beta toolchain will be broken in the same way it's broken in current nightly (https://github.com/rust-lang/rust/issues/68591).

**Comment by pietroalbini** [comment]
@bors retry

Let's handle beta backports the normal way.

**Comment by JohnTitor** [comment]
It seems like to be blocked by toolstate (CI says `rustfmt` is `test-fail`).

**Comment by pietroalbini** [comment]
Yep, I tried it locally before retrying but it passed :/

@rust-lang/wg-rustfmt do you have any idea on the cause of the failure?

**Comment by JohnTitor** [comment]
At a glance, there're warnings which are only shown here, like:

```
Warning: can't set `report_todo = Always`, unstable features are only available in nightly channel.
```

So, `test_valid_license_template_path` seems failed.

File: src/stage0.txt [hunk 1]
```diff
@@ -12,14 +12,14 @@
...
-cargo: beta
+date: 2020-01-27
+rustc: 1.41.0
+cargo: 0.42.0
 
 # We use a nightly rustfmt to format the source because it solves some bootstrapping
 # issues with use of new syntax in this repo. If you're looking at the beta/stable branch, this key should be omitted,
 # as we don't want to depend on rustfmt from nightly there.
-rustfmt: nightly-2019-12-18
+#rustfmt: nightly-2019-12-18
```
Comment (JohnTitor) [inline src/stage0.txt]: I'm really not sure but this may cause the failure here? (Yeah, the comment says it should be omitted though.)

Comment (Mark-Simulacrum) [reply src/stage0.txt]: No, this should not be responsible. The rustfmt here is purely for x.py fmt, it doesn't affect testing at all (we build a rustfmt from scratch from that).

Comment (JohnTitor) [reply src/stage0.txt]: Oh, didn't know that. So it shouldn't be the cause...

**Comment by mati865** [comment]
Failing rustfmt tests should've been ignored on beta: https://github.com/rust-lang/rustfmt/blob/cc56957656ff9b14168805104f901b19550cea07/rustfmt-core/rustfmt-config/src/lib.rs#L481

There were few changes to `src/tools/compiletest` recently. Maybe it doesn't set `CFG_RELEASE_CHANNEL` environment variable anymore?

**Comment by Mark-Simulacrum** [comment]
Compiletest wasn't used for these tests AFAICT, they're just regular `cargo test` tests?

https://github.com/rust-lang/rustfmt/blob/1838235248bbde125bbc6c12fb493e1979088bb5/src/config/mod.rs#L477-L482 is the source for the failing test_valid_license_template_path test, which does not have the gate you noted? I think you are pointing at the wrong commit unless I'm missing something. rustfmt on beta is at rust-lang/rustfmt@1838235

We probably do need that annotation though...

**Comment by Mark-Simulacrum** [comment]
I have no idea which of the system tests are failing though, nor do I know how to find out. rustfmt doesn't appear to log anywhere which one actually failed, which is super unhelpful. It could be more missing gating on instability, though.

cc @topecongiro

**Comment by mati865** [comment]
Yeah, you are right.
Failing tests:
```
config::test::test_valid_license_template_path
test::system_tests
```

**Comment by Mark-Simulacrum** [comment]
Okay, I bumped the rustfmt submodule to include a commit which backports rust-lang/rustfmt@f82171a4b0867cfa85c9208668c784c0ea69ea28 (my impression is that this fixes the test failures). I also confirmed Cargo is sufficiently recent that we should avoid the main infinite loop bug.

@bors r+ p=100

**Pull request merged**